staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset.go

```diff
@@ -281,39 +296,13 @@ func (qs *queueSet) StartRequest(ctx context.Context, width *fqrequest.Width, ha
 	// request from that queue.
 	qs.dispatchAsMuchAsPossibleLocked()
 
-	// ========================================================================
-	// Step 3:
-
-	// Set up a relay from the context's Done channel to the world
-	// of well-counted goroutines. We Are Told that every
-	// request's context's Done channel gets closed by the time
-	// the request is done being processed.
-	doneCh := ctx.Done()
-
-	// Retrieve the queueset configuration name while we have the lock
-	// and use it in the goroutine below.
-	configName := qs.qCfg.Name
-
-	if doneCh != nil {
-		qs.preCreateOrUnblockGoroutine()
-		go func() {
-			defer runtime.HandleCrash()
-			qs.goroutineDoneOrBlocked()
-			_ = <-doneCh
-			// Whatever goroutine unblocked the preceding receive MUST
-			// have already either (a) incremented qs.counter or (b)
-			// known that said counter is not actually counting or (c)
-			// known that the count does not need to be accurate.
-			// BTW, the count only needs to be accurate in a test that
-			// uses FakeEventClock::Run().
-			klog.V(6).Infof("QS(%s): Context of request %q %#+v %#+v is Done", configName, fsName, descr1, descr2)
-			qs.cancelWait(req)
-			qs.goroutineDoneOrBlocked()
-		}()
-	}
 	return req, false
 }
 
+func normalPromiseMakerMaker(qs *queueSet) queueSetPromiseMaker {
```

Comment:
simplePromiseMakerFactory maybe?

Reply:
The distinction between this implementation and the other one is not primarily about simple vs complex but rather normal code vs testing code.

Reply:
I've never heard "normal" being used in such context.

defaultPromiseFactory or sth [and call the other testPromiseFactory]

I just don't know how to interpret normal - it doesn't mean anything useful.

Reply:
"default" suggests to me that ordinary code might want a non-default value.  I am looking for a word that more clearly contrasts with test code.  How about "regular"?

Reply:
or "ordinary"?